File: department-service/src/main/java/pl/piomin/services/department/model/Department.java
package pl.piomin.services.department.model;

import io.micronaut.serde.annotation.Serdeable;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@ToString
@NoArgsConstructor
@Serdeable
public class Department {

	private Long id;
	private Long organizationId;
	private String name;
	private List<Employee> employees = new ArrayList<>();

	public Department(Long organizationId, String name) {
		super();
		this.organizationId = organizationId;
		this.name = name;
	}

}

File: department-service/src/main/kotlin/pl/piomin/services/department/model/Department.kt
package pl.piomin.services.department.model;

import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class Department(@Id var id: Long, var organizationId: Long, var name: String) {

    @Transient
    var employees: MutableList<Employee> = mutableListOf()

}
